src/init.cpp

```diff
@@ -1841,6 +1841,55 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
         }
     }
 
+#ifdef __APPLE__
+    // Warn about exFAT filesystem usage on macOS
+    struct PathCheck {
+        fs::path path;
+        std::string_view description;
+        PathCheck(const fs::path& p, std::string_view d) : path(p), description(d) {}
+    };
+
+
+    fs::path data_dir = args.GetDataDirNet();
+    fs::path blocks_dir = args.GetBlocksDirPath();
+    std::vector<PathCheck> paths{{data_dir, "data directory"}};
+    // Don't mention blocks dir if it's inside datadir
```

Comment:
Does macOS not allow mounting subdirectories? (Maybe a bind mount) eg, maybe datadir isn't exFAT, but blocks is?

Reply:
Agree. I removed this "optimisation" in eb68bb48ab2bb163f20dc7430d0401d88ca43f4a

We now just check both directories to avoid missing any as you describe (and if they are renamed as per https://github.com/bitcoin/bitcoin/pull/31453#discussion_r2171706368 )